rich/syntax.py

```diff
@@ -448,7 +484,7 @@ def highlight(
 
                 def line_tokenize() -> Iterable[Tuple[Any, str]]:
                     """Split tokens to one per line."""
-                    assert lexer
+                    assert lexer  # required to make MyPy happy
```

Comment:
For future devs, the comment should probably explain why. Suggest "we know lexer is not None at this point" or words to that effect.